src/slotscheck/discovery.py

```diff
@@ -299,15 +300,19 @@ def _is_package(p: AbsPath) -> bool:
     return p.is_dir() and (p / _INIT_PY).is_file()
 
 
-def find_modules(p: AbsPath) -> Iterable[ModuleLocated]:
+def find_modules(
+    p: AbsPath, python_path: Optional[FrozenSet[AbsPath]] = None
+) -> Iterable[ModuleLocated]:
     "Recursively find modules at given path. Nonexistent Path is ignored"
+    if python_path is None:
```

Comment:
seems this `if` is only triggered on the first call. Can you make the argument non-optional to prevent this check every call? Minor inconvenience to callers to do `frozenset(map(Path, sys.path))` themselves.

Reply:
Factored out an internal helper function instead so I don't need to update every caller of `find_modules`, and also in case we'll want to change the algorithm in the future.